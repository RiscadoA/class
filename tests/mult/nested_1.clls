proc main() {
	cut {
		println("send x");
		send x(z.
			println("recv z");
			recv z(w);
			println("wait w");
			wait w;
			println("close z");
			close z
		);
		println("wait x");
		wait x;
		println("end");
		()
		|x: recv send close; wait; close|
		println("recv x");
		recv x(y);
		println("send y");
		send y(t. println("close t"); close t);
		println("wait y");
		wait y;
		println("close x");
		close x
	}
};;
