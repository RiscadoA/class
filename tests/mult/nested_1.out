send x
recv x
send y
recv z
close t
wait w
close z
wait y
close x
wait x
end
